.github/actions/setup-solana/action.yml

```
@@ -0,0 +1,71 @@
+# adjusted from https://github.com/metaDAOproject/setup-solana/blob/3769a7937e9897f27805e9665507b7b754cbc7fc/action.yml
+name: "Setup Solana"
+description: "Install the Solana Tool Suite."
+inputs:
+    solana-cli-version:
+        description: "Version of Solana CLI to use"
+        required: false
+        default: "stable"
+outputs:
+    installed-solana-cli-version:
+        description: "The version of Solana CLI that was installed"
+        value: ${{ steps.solana-release.outputs.install_version }}
+runs:
+    using: "composite"
+    steps:
+        - name: download scripts
+          run: |
+              mkdir -p scripts/semver_bash
+              curl -sSfL https://raw.githubusercontent.com/anza-xyz/agave/master/ci/channel-info.sh -o scripts/channel-info.sh
+              curl -sSfL https://raw.githubusercontent.com/anza-xyz/agave/master/ci/semver_bash/semver.sh -o scripts/semver_bash/semver.sh
+          shell: bash
+
+        - name: Determine Solana CLI version
+          id: solana-release
+          run: |
+              chmod +x scripts/channel-info.sh
+              output=$("scripts/channel-info.sh")
+              read -r BETA_CHANNEL_LATEST_TAG STABLE_CHANNEL_LATEST_TAG <<< "$output"
+
+              if [[ "${{ inputs.solana-cli-version }}" =~ ^[a-zA-Z] ]]; then
+                if [[ ${{ inputs.solana-cli-version }} =~ ^(edge|stable|beta)$ ]]; then
+                  if [[ ${{ inputs.solana-cli-version }} == "edge" ]]; then
+                    echo "Using Solana CLI edge channel"
+                    echo "install_version=edge" >> $GITHUB_OUTPUT
+                  elif [[ ${{ inputs.solana-cli-version }} == "stable" ]]; then
+                    echo "Using Solana CLI stable channel $install_version"
+                    echo "install_version=$STABLE_CHANNEL_LATEST_TAG" >> $GITHUB_OUTPUT
```

```suggestion
                    echo "install_version=$STABLE_CHANNEL_LATEST_TAG" >> $GITHUB_OUTPUT
                    echo "Using Solana CLI stable channel $install_version"
```
install_version was used before set